### PR TITLE
Build system cleanup

### PR DESCRIPTION
**Pull request by pspacek**
This is first step in build system refactoring effort.

This patch set contains cleanup patches for daemons/configure.ac.
After the cleanup, the file will be "promoted" to top-level configure.ac
and merged with other configure.ac files in subdirectories.

I did not touch other configure.ac files on purpose as these mostly duplicate
daemons/configure.ac and will be simply dropped later on.

From functional perspective, there should not be any visible changes.
FreeIPA should build as before, using the same horrible Makefile.

### CODE REVIEW

File: daemons/configure.ac [hunk 1]
```diff
@@ -91,7 +91,7 @@ dnl ---------------------------------------------------------------------------
 
 AC_MSG_NOTICE([Checking for Python])
 have_python=no
-AM_PATH_PYTHON(2.3)
+AM_PATH_PYTHON(2.7)
```
Comment (mbasti-rh) [inline daemons/configure.ac]: ACK for this

**Comment by mbasti-rh** [comment]
Works for me, Not pushing yet to give time to others to disagree

**Comment by mbasti-rh** [comment]
Needs rebase

**Comment by abbra** [comment]
I think all these fix commits are just fine.

**Comment by mbasti-rh** [comment]
Fixed upstream
master:
https://fedorahosted.org/freeipa/changeset/152ed75ff6e55489e27c201b27ee6eb5c4db0480
https://fedorahosted.org/freeipa/changeset/651f1acac8212346fb579ff8911b43a3faf3867b
https://fedorahosted.org/freeipa/changeset/7b801682a64b2ba132ba9892831f8c3f5a51caf8
https://fedorahosted.org/freeipa/changeset/01072fc8f2833e9316f534875c75714803384377
https://fedorahosted.org/freeipa/changeset/2ea66483791ef060c54f525786fbac2730813bbc
https://fedorahosted.org/freeipa/changeset/8b458ce4de9cb0e072849c4fbd9257377cd4a654
https://fedorahosted.org/freeipa/changeset/7de597425f8fb6d8c5cadc8fc039729368c43ddf

**Pull request closed without merging**